plugins/dell-dock/fu-plugin-dell-dock.c

```diff
@@ -184,6 +184,24 @@ fu_plugin_device_registered (FuPlugin *plugin, FuDevice *device)
 		return;
 	}
 
+	/* online activation is mutually exclusive between usb4 and ec */
+	if (g_strcmp0(fu_device_get_plugin(device), "dell_dock") == 0 &&
+	    FU_IS_DELL_DOCK_USB4(device) &&
+	    fu_device_has_flag(device, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION)) {
+		GPtrArray *devices = fu_plugin_get_devices(plugin);
+		for (guint i = 0; i < devices->len; i++) {
+			FuDevice *device_tmp = g_ptr_array_index(devices, i);
+			if (FU_IS_DELL_DOCK_EC(device_tmp) &&
+			    fu_device_has_flag(device_tmp, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION)) {
+				fu_device_remove_flag(FU_DEVICE(device_tmp),
+						      FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION);
+				g_debug("activate for %s is suspended by %s",
```

Comment:
How does this flag ever come back now? I wonder if we are actually talking about inhibited but for activation not update.

Reply:
But actually looking at it, I see that might be problematic because inhibitition is usually called when something is put into needs-activate state already.  I think what you've done is OK.